partiql-planner/src/main/kotlin/org/partiql/planner/catalog/Catalog.kt

```diff
@@ -0,0 +1,56 @@
+package org.partiql.planner.catalog
+
+/**
+ * Catalog interface for access to tables and routines.
+ *
+ * Related
+ *  - Iceberg — https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/catalog/Catalog.java
+ *  - Calcite — https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/schema/Schema.java
+ */
+public interface Catalog {
+
+    /**
+     * Returns the catalog name.
+     */
+    public fun getName(): String
+
+    /**
+     * Get a table by name.
+     *
+     * @param name  The case-sensitive [Table] name.
+     * @return The [Table] or null if not found.
+     */
+    public fun getTable(name: Name): Table? = null
+
+    /**
+     * List top-level tables.
+     */
+    public fun listTables(): Name? = null
+
+    /**
+     * List all tables under this namespace.
+     *
+     * @param namespace
+     */
+    public fun listTables(namespace: Namespace): List<Name> = emptyList()
+
+    /**
+     * List top-level namespaces from the catalog.
+     */
+    public fun listNamespaces(): Collection<Namespace> = emptyList()
+
+    /**
+     * List all child namespaces from the namespace.
+     *
+     * @param namespace
+     */
+    public fun listNamespaces(namespace: Namespace)
```

Comment:
Return type seems off for this function perhaps should be a `Collection<Namespace>`?

```suggestion
    /**
     * List all child namespaces from the namespace.
     *
     * @param namespace
     */
    public fun listNamespaces(namespace: Namespace): Collection<Namespace> = emptyList()
```